### PR TITLE
Feature 156 UI cleanup and components

### LINKED ISSUE #156: UI cleanup and components

### PR DESCRIPTION
**Pull request by Kaemppi**
Resolves #156

### CODE REVIEW

File: src/components/layout/app-layout.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { NavigationProp } from "@react-navigation/native";
+import React from "react";
```
Comment (Daniilmass) [inline src/components/layout/app-layout.tsx:2]: What is the reason for using react functional component here?

**Review by Daniilmass — CHANGES_REQUESTED** [review]
Are there any benefits in using the react functional components in this application? Just curious.

File: src/components/generic/progress-bar-accordion.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  title: string;
+  progress: number;
+}
+
+/**
+ * Component for rendering progress bar indicator with accordion content
+ */
+const ProgressBarAccordion: React.FC<Props> = ({ title, children, progress }) => {
+  
+  const [ open, setOpen ] = useState(false);
```
[the remark this answers is not in the document]
Comment (Kaemppi) [reply src/components/generic/progress-bar-accordion.tsx:20]: The company code style seems to encourage to use it such way to help readability.

[the remark this answers is not in the document]
Comment (Daniilmass) [reply src/components/generic/progress-bar-accordion.tsx:20]: Oh! I thought it applies to { ... } not the [ ... ]

File: src/components/generic/rounded-button.tsx [hunk 3]
```diff
@@ -40,7 +39,11 @@ class RoundedButton extends React.Component<Props, State> {
    * Component render method
    */
   public render = () => {
-    const { icon, title, description } = this.props;
+    const {
+      icon,
+      title,
+      description
+    } = this.props;
```
[the remark this answers is not in the document]
Comment (Kaemppi) [reply src/components/generic/rounded-button.tsx:46]: I have used it when it goes over two to increase readability

File: src/components/layout/app-layout.tsx (hunk 1 above)
Comment (Kaemppi) [reply src/components/layout/app-layout.tsx:2]: We have preferred using the functional format in simple components. Functionals are also pretty useful in general.

**Pull request merged**